lib/rspec/core/memoized_helpers.rb

```diff
@@ -123,14 +123,28 @@ def is_expected
 
       # @private
       def __memoized
-        @__memoized ||= {}
+        @__memoized ||= Memoized.new
```

Comment:
I was thinking of an implicit subject. With `let`, the user calls it before running the test, so we know it needs to be set, but with implicit `subject`, the user doesn't have to call anything ahead of time.

```ruby
RSpec.describe Object do
  it 'is an Object' do
    Thread.new { should_not be_a_kind_of String }
    should be_a_kind_of Object
  end
end
```